compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -528,10 +528,14 @@ class TreePickler(pickler: TastyPickler) {
             }
           }
         case Bind(name, body) =>
-          registerDef(tree.symbol)
+          val sym = tree.symbol
+          registerDef(sym)
           writeByte(BIND)
           withLength {
-            pickleName(name); pickleType(tree.symbol.info); pickleTree(body)
+            pickleName(name)
+            pickleType(sym.info)
+            pickleTree(body)
+            pickleFlags(sym.flags &~ Case, sym.isTerm)
```

Comment:
Do we need to allow other modifiers like annotations? Can these be on the binding or are they always on the type?

Reply:
They should be on the type.  Are there other flags on bound symbols except Implicit or Given?

Reply:
Not now. But I could imagine there could be a Synthetic flag added to some of those.

It may also make sense to have `Erased` and `Inline` in the future. Such as in #11598.